### PR TITLE
51 - Create like model

### LINKED ISSUE #51: Create Like Model

### PR DESCRIPTION
**Pull request by h-yung**
Created LikeModel inside /models with reference to likeController.like and like.dao.js.

About the `planId` setup in the model. My understanding is:
- planId would be a hexadecimal string (one of the req.params) 
- string is passed to ObjectId() but essentially generates a new ObjectId for the Like that matches the planId

**To confirm**: LikeModel pertains to a user story where a user is expressing interest in and adding/associating a particular plan to their account?
**Warning**: Some hijinks ensued trying to get server running, so may include odd flags around `package.json` changes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "@aws-sdk/client-ses": "^3.76.0",
     "@types/express": "^4.17.13",
     "@types/node": "^18.7.16",
+    "all": "^0.0.0",
```
Comment (jpatrici) [inline package.json]: can this be deleted then, since we dont need it?

File: src/app/models/LikeModel.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ _id: mongodb.ObjectId;
+ email: string;
+ planId: mongodb.ObjectId; 
+ createdAt: Date;
+
+ constructor(
+    email = '',
+    planId = '', 
+    createdAt = new Date(),
+
```
Comment (jpatrici) [inline src/app/models/LikeModel.ts]: extra line not needed and can be deleted

**Review by jpatrici — APPROVED** [review]

**Review by RDiMag — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (theodorusandi) [reply package.json]: yes. I don't remember installing this one.

File: src/app/models/LikeModel.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+class LikeModel {
+ _id: mongodb.ObjectId;
+ email: string;
+ planId: mongodb.ObjectId; 
+ createdAt: Date;
+
+ constructor(
+    email = '',
+    planId = '', 
+    createdAt = new Date(),
```
Comment (theodorusandi) [inline src/app/models/LikeModel.ts]: we don't allow user to enter the createdAt date themselves so it should not be passed in as a param.

File: package.json (hunk 1 above)
Comment (h-yung) [reply package.json]: removed in latest commit to branch! Was accidental

File: src/app/models/LikeModel.ts (hunk 2 above)
Comment (h-yung) [reply src/app/models/LikeModel.ts]: removed in latest commit to branch

File: src/app/models/LikeModel.ts (hunk 3 above)
Comment (h-yung) [reply src/app/models/LikeModel.ts]: Can you check latest commit (pushed just now)? original included as I was modeling after ClassModel which has createdAt generated/assigned as a param.

**Review by Marcellia — APPROVED** [review]

**Pull request merged**